python/GafferUI/NumericWidget.py

```diff
@@ -285,6 +285,14 @@ def __setValueInternal( self, value, reason ) :
 			self.__numericType = numericType
 			self._qtWidget().setValidator( _ExpressionValidator( self._qtWidget(), self.__numericType ) )
 
+		# If the new value and the current value are the same, don't set the value.
+		# We compare the text representations to account for floating point text formatting.
+		try :
+			if self.getText() and NumericWidget.valueToString( value ) == str( self.getValue() ) :
+				return
+		except ValueError :
+			pass
```

Comment:
The comment on L301 is relevant here - we shouldn't be doing an early-out if this is the start or end of a drag. Maybe we want to be replacing the check on L300 rather than adding an additional check?

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5370/commits/b8798235ff02f470670eca7716c9e0e06e48020d.

I also removed my exception handling - I was trying to catch the exception raised when you try to up/down key on a field with a half-finished arithmetic expression. But that exception was being raised elsewhere so it's unrelated and this meshes better with the existing code.